juju/juju
# Aws instance profile warning

With the introduction of instance profile code there is a potentially error for users when destroying controllers that have a very limited permission set in use because 2.9 now requires more permissions then it did previously. This PR introduces the new checks as warning's should this case arise.

## Checklist


## QA steps

Create a aws credential user with the policy found [here](https://discourse.charmhub.io/t/juju-aws-permissions/5307). From this policy remove the `"iam:ListInstanceProfiles"` and `"iam:ListRoles"` actions.

Add the newly created AWS user to your local Juju credentials and bootstrap a controller.

Kill the controller with:
`juju kill-controller -t0 -y`

You should see warnings in the output about the lack of permissions and manual cleanup may be required.

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1954915
